client/my-sites/checkout/controller.jsx

```diff
@@ -175,6 +189,19 @@ export function checkout( context, next ) {
 
 	const product = getProductSlugFromContext( context );
 
+	if ( ( isJetpackPlanSlug( product ) || isJetpackProductSlug( product ) ) && isLoggedOut ) {
+		// Redirect to the siteless checkout page
+		const redirectUrl = addQueryArgs(
+			{
+				connect_after_checkout: true,
+				from_site_slug: context.query.site,
+				admin_url: context.query.redirect_to.split( '?' )[ 0 ],
+			},
+			context.path.replace( /checkout\/[^?/]+\//, 'checkout/jetpack/' )
+		);
+		page( redirectUrl );
+	}
```

Comment:
Do you think it would work to have this logic in the new middleware function?

Reply:
It would, I just didn't wanna rely on always having the same order when calling the middleware functions. Even though I feel like that's a condition that would better fit a middleware, if we change the order of the middleware functions in any way, it will stop working.

Reply:
But I guess people are usually cautious when doing stuff like that, so I'm pushing the changes.